src/Common/src/System/Marvin.cs

```diff
@@ -57,7 +57,7 @@ public static long ComputeHash(ReadOnlySpan<byte> data, ulong seed)
                     break;
 
                 case 2:
-                    p0 += 0x800000u | MemoryMarshal.Cast<byte, ushort>(data)[0];
+                    p0 += 0x800000u | ((uint)data[0]) << 8 | data[1];
```

Comment:
there is also one more: https://github.com/dotnet/corefx/pull/28110/files#diff-cd635355a880d8cef033635b10e2e4deR64